results/cmd/api/main_test.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	pb "github.com/tektoncd/experimental/results/proto/proto"
+)
+
+const (
+	address = "localhost:50051"
+)
+
+// Test functionality of Server code
+func TestCreateTaskRun(t *testing.T) {
+	// Set up a connection to the server.
+	tmpfile, err := ioutil.TempFile("", "testdb")
+	if err != nil {
+		t.Fatalf("failed to create temp file for db: %v", err)
+	}
+
+	// clean up at the end of test
+	defer os.Remove(tmpfile.Name())
+	testDB, err := setupTestDB(tmpfile.Name())
+	defer testDB.Close()
+	if err != nil {
+		t.Fatalf("failed to create temp file for db: %v", err)
+	}
+
+	// connect to fake server and do testing
+	srv := &server{db: testDB}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	objectmeta := &pb.ObjectMeta{Uid: "123459", Name: "mytaskrun", Namespace: "default"}
+	r, err := srv.CreateTaskRun(ctx, &pb.CreateTaskRunRequest{TaskRun: &pb.TaskRun{ApiVersion: "1", Metadata: objectmeta}})
+	if err != nil {
+		t.Fatalf("could not create taskrun: %v", err)
+	}
+	t.Logf("Created taskrun: %s", r.String())
+}
+
+// setupTestDB set up a temporary database for testing
+func setupTestDB(n string) (*sql.DB, error) {
+	// Connect to sqlite DB.
+	db, err := sql.Open("sqlite3", n)
+	if err != nil {
+		log.Fatalf("failed to open the results.db: %v", err)
```

Comment:
Prefer testing.T over log where possible (e.g. log.Fatal -> t.Fatal). It generally gives you richer information in test results, and allows you to continue to run other tests even if this one particular test fails.

Reply:
done